tests/test_schedules.py

```diff
@@ -427,6 +427,64 @@ def foo(n: size, x: i8):
     foo = expand_dim(foo, 'a : i8', 'n', 'i')
     assert str(foo) == golden
 
+def test_divide_dim_1(golden):
+    @proc
+    def foo(n: size, m: size, A : R[n + m + 12]):
+        x : R[n, 12, m]
+        for i in seq(0, n):
+            for j in seq(0, 12):
+                for k in seq(0, m):
+                    x[i, j, k] = A[i + j + k]
+
+    foo = divide_dim(foo, 'x', 1, 4)
```

Comment:
What will happen if there are multiple 'x' buffers in this proc? It seems the API does not loop over multiple Syms?

Reply:
All of the scheduling operations are now designed to apply exactly once.  Handling multiple applications of a scheduling operation should now be handled using higher-order combinators.  This will be even easier once step 2 of the API update is done, since then you can do something like `p.find_all(pattern)` to get a list of cursors, and then repeat an operation for each cursor in the list.